rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -219,21 +223,23 @@ void RSocketStateMachine::closeFrameTransport(
   }
 
   if (resumeCallback_) {
-    resumeCallback_->onResumeError(ConnectionException(
+    auto resumeCallback = std::move(resumeCallback_);
+    resumeCallback->onResumeError(ConnectionException(
```

Comment:
The bug fix here:

Say resumption fails for some reason and we call closeFrameTransport().  onResumeError() will then fulfill a promise.  Depending on scheduling, immediately after fulfilling the promise the thread will destroy RSocketClient (before resumeCallback_.reset() gets executed).  The destruction of RSocketClient will lead to invoking resumeCallback_->onResumeError() if resumeCallback_ is not null (which will be the case since resumeCallback_.reset() never got to run).

Reply:
nit: move the move to the if check :)